adobe/da-live
# Allow authoring external links

## Description

- Before: https://main--da-live--mokimo.hlx.page/#/aemsites/da-loves-ue/drafts/osahin/hackathon
- After: https://extend-file-browser--da-live--mokimo.hlx.page/#/aemsites/da-loves-ue/drafts/osahin/hackathon

Adds a new `.link` filetype that can be added 

## Related Issue

This has been discussed in person at the DA Hackathon

## Motivation and Context

With the content overlay feature, we blend DA and SP content. We want to give authors a way to link back to Sharepoint from within DA to provide a consistent experience

## How Has This Been Tested?

- Created a `.link` file
- Renamed the `.link` file
- Deleted the `.link` file (there was a general issue with deleting things...*)
- Open the created `.link` file
- Ensure [content.da.live](https://content.da.live/aemsites/da-loves-ue/drafts/osahin/hackathon/link-to-adobe.link) works
- Create all other types of doc/media/sheet and see that it all works
- Creating a link programmatically
```js
const DA_API = 'https://admin.da.live/source';
const ORG = 'aemsites';
const REPO = 'da-loves-ue';

(async () => {
  const body = new FormData();
  const data = new Blob([JSON.stringify({ externalUrl: "https://adobe.com" })], { type: 'text/html' });
  body.set('data', data);

  // const headers = { Authorization: `Bearer {{YOUR_TOKEN_HERE}}` };
  const opts = { method: 'POST', body };

  const path = `${DA_API}/${ORG}/${REPO}/drafts/osahin/node-demo.link`;
  const resp = await fetch(path, opts);
  console.log(resp.status);
})();
```

## Screenshots (if appropriate):

1. Creating a new "link"
<img width="263" alt="Screenshot 2024-12-04 at 15 34 41" src="https://github.com/user-attachments/assets/0f345d39-f894-408c-abd1-5a1da5db0ebf">

2. Inputs for that
<img width="583" alt="Screenshot 2024-12-04 at 15 34 46" src="https://github.com/user-attachments/assets/c61f5733-27a2-4cdb-90b2-c0a3fa683938">

3. List view
<img width="1105" alt="Screenshot 2024-12-04 at 15 34 55" src="https://github.com/user-attachments/assets/66ca271e-f48a-4ba3-85fb-4d6b8d773874">

4. The whole thing
<img width="1393" alt="Screenshot 2024-12-04 at 15 34 58" src="https://github.com/user-attachments/assets/a8db68ee-8ef4-407a-946e-3dd85fa08d56">

## Types of changes


## Checklist:




- [ ] My change requires a change to the documentation. ... **does it?**
